packages/share_plus/share_plus/android/src/main/kotlin/dev/fluttercommunity/plus/share/MethodCallHandler.kt

```diff
@@ -14,41 +15,42 @@ internal class MethodCallHandler(
     override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
         expectMapArguments(call)
 
-        // The user used a *WithResult method
-        val isResultRequested = call.method.endsWith("WithResult")
         // We don't attempt to return a result if the current API version doesn't support it
         val isWithResult =
-            isResultRequested && Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP_MR1
+            Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP_MR1
 
-        if (isWithResult && !manager.setCallback(result)) return
+        if (isWithResult)
+            manager.setCallback(result)
```

Comment:
this change fixes any possible deadlocks, since `setCallback` will force any pending call to be returned with an "unavailable" `ShareResult`